tests/reorder_python_imports_test.py

```diff
@@ -524,6 +524,47 @@ def test_replace_module_imported_asname():
     assert ret == 'import queue as Queue\n'
 
 
+def test_replace_module_imported_with_nested_replacement():
+    ret = fix_file_contents(
+        'from six.moves.urllib import parse\n',
+        imports_to_replace=[(['six', 'moves', 'urllib', 'parse'], ['urllib', 'parse'], '')],
+    )
+    assert ret == 'from urllib import parse\n'
+
+
+def test_replace_module_imported_with_nested_replacement_asname():
+    ret = fix_file_contents(
+        'from six.moves.urllib import parse as urllib_parse\n',
+        imports_to_replace=[(['six', 'moves', 'urllib', 'parse'], ['urllib', 'parse'], '')],
+    )
+    assert ret == 'from urllib import parse as urllib_parse\n'
+
+
+@pytest.mark.xfail(reason='TODO')
```

Comment:
Wasn't sure exactly how to implement this rewrite. We split imports before the replace pass so AFAIK this should be safe to do, but just didn't know what code to write to patch up the ast_obj.names/list of aliases (which should just be the one entry).

Reply:
is there a reason this one doesn't work? it looks like it should

Reply:
Figured out how to implement this, push https://github.com/asottile/reorder_python_imports/pull/239/commits/4fa54ac7b35902902af393d8d5cbec195165bdc6 to implement it.